### PR TITLE
Tarun, Gopi | MOBN-1628 | Add support to create surgical blocks across days

### PR DESCRIPTION
**Pull request by tarunkumar-tw**

### CODE REVIEW

File: ui/app/ot/directives/otCalendarSurgicalBlock.js [hunk 1]
```diff
@@ -18,7 +18,10 @@ angular.module('bahmni.ot')
                     left: $scope.weekOrDay === 'week' ? getLeftPositionForSurgicalBlock() : 0,
                     color: getColorForProvider(),
                     appointmentHeightPerMin: (surgicalBlockHeight - heightForSurgeonName) / Bahmni.Common.Util.DateUtil.diffInMinutes(
-                        $scope.surgicalBlock.startDatetime, $scope.surgicalBlock.endDatetime)
+                        moment($scope.surgicalBlock.startDatetime).toDate() < getCalendarStartDateTime($scope.viewDate)
+                            ? getCalendarStartDateTime($scope.viewDate) : $scope.surgicalBlock.startDatetime,
+                        getCalendarEndDateTime($scope.viewDate) < moment($scope.surgicalBlock.endDatetime).toDate()
+                            ? getCalendarEndDateTime($scope.viewDate) : $scope.surgicalBlock.endDatetime)
```
Comment (binduak) [inline ui/app/ot/directives/otCalendarSurgicalBlock.js:24]: @tarunkumar-tw can we refactor the common code into a method ?

File: ui/app/ot/directives/otCalendarSurgicalBlock.js [hunk 2]
```diff
@@ -50,12 +53,16 @@ angular.module('bahmni.ot')
             };
             var getHeightForSurgicalBlock = function () {
                 return Bahmni.Common.Util.DateUtil.diffInMinutes(
-                        $scope.surgicalBlock.startDatetime, $scope.surgicalBlock.endDatetime) * surgicalBlockHeightPerMin;
+                    moment($scope.surgicalBlock.startDatetime).toDate() < getCalendarStartDateTime($scope.viewDate)
+                        ? getCalendarStartDateTime($scope.viewDate) : $scope.surgicalBlock.startDatetime,
+                    getCalendarEndDateTime($scope.viewDate) < moment($scope.surgicalBlock.endDatetime).toDate()
+                        ? getCalendarEndDateTime($scope.viewDate) : $scope.surgicalBlock.endDatetime) * surgicalBlockHeightPerMin;
```
Comment (binduak) [inline ui/app/ot/directives/otCalendarSurgicalBlock.js:59]: @tarunkumar-tw ^^

File: ui/app/ot/directives/otCalendarSurgicalBlock.js (hunk 1 above)
Comment (tarunkumar-tw) [reply ui/app/ot/directives/otCalendarSurgicalBlock.js]: Done.

File: ui/app/ot/directives/otCalendarSurgicalBlock.js (hunk 2 above)
Comment (tarunkumar-tw) [reply ui/app/ot/directives/otCalendarSurgicalBlock.js]: Done

**Review by binduak — APPROVED** [review]

**Pull request merged**